docs/topics/genericgateway.md

```diff
@@ -47,24 +64,105 @@ When calling the Generic Gateway `simpleevent` endpoint, you must include a cust
 }
 ```
 
-`Ref` and `commit` values are used to configure the specific revision that Brigade will pull from your repository, if you're using one. If both values are missing, then your Build's `ref` will be set to `master`. If you do not wish to provide any payload, just include an empty JSON object (like `{}`).
+You can use curl to test it:
 
-Last but not least, here is a sample Brigade.js file that could be used as a base for your own scripts that respond to Generic Gateway's `webhook` event. This script will echo the name of your project and 'webhook'.
+```bash
+curl --header "Content-Type: application/json" \
+  --request POST \
+  --data '{
+    "ref": "refs/heads/changes",
+    "commit": "b60ad9543b2ddbbe73430dd6898b75883306cecc",
+    "key1": "value1",
+    "key2": "value2"
+}' \
+  http://localhost:8000/simpleevent/PROJECT_ID/SECRET
+```
+
+If you do not wish to provide any payload, you can send empty POST data or just an empty JSON object (`{}`). 
+
+---
+**NOTE**
+
+If you plan to use this type of event with source control, you should specify `ref` and `commit` values. Those are used to configure the specific revision that Brigade will pull from your repository. If both values are missing, your Build's `ref` will be set to `master`. Of course, you can also provide any other values you need.
+
+---
+
+### Calling the CloudEvent endpoint
+
+When calling the Generic Gateway `cloudevent` endpoint, you must include a valid [0.2 CloudEvent](https://github.com/cloudevents/spec/blob/v0.2/spec.md) message such as:
+
+```json
+{
+    "type":   "com.example.file.created",
+    "source": "/providers/Example.COM/storage/account#fileServices/default/{new-file}",
+    "id":     "ea35b24ede421",
+    "specversion": "0.2",
+    "data": {
+        "ref": "refs/heads/changes",
+        "commit": "63c09efb6eb544f41a48901a6d0cc6ddfa4adb28",
+        "key1": "value1",
+        "key2": "value2"
+  }
+}
+```
+
+You can use curl to test it:
+
+```bash
+curl --header "Content-Type: application/json" \
+  --request POST \
+  --data '{ "type":   "com.example.file.created",
```

Comment:
If desired, we could reference the json above (if placed in file) and remove duplication in this curl cmd using `--data-binary @example.json` or `--data "$(cat example.json)"`

Reply:
Hm, in this case wouldn't we lose the benefit to simply copy/paste this command (for testing purposes)?

Reply:
You're right, we would lose that ability.  Let's keep it 👍